packages/core/src/Utils/Datadog/index.ts

```diff
@@ -45,9 +45,14 @@ const getConfigValues = (environment: string) => {
  * @returns {void}
  * **/
 const initDatadog = (is_datadog_enabled: boolean) => {
-    if (!is_datadog_enabled) {
+    if (is_datadog_enabled) {
         if (window.DD_RUM) {
-            window.DD_RUM = undefined;
+            datadogRum.setTrackingConsent('granted');
+            return;
+        }
+    } else {
+        if (window.DD_RUM) {
+            datadogRum.setTrackingConsent('not-granted');
         }
         return;
     }
```

Comment:
If `datadog` is `enabled` in Firebase and `datadogRum` is already initialised, we `"grant"` tracking consent and `return` from the `function`. This prevents re-initialisation, which would otherwise cause an error. If `datadogRum` is not initialised yet, we continue execution further down the function to initialise it.

On the other hand, if `datadog` is not `enabled` in Firebase, we check if `datadogRum` is already initialised. If it is, we revoke the tracking consent and `return` from the `function`. If it's not initialised yet, then we still just `return` from the `function`